sha1/src/checked.rs

```diff
@@ -0,0 +1,386 @@
+//! Collision checked Sha1.
+//!
+//! General techniques and implementation are based on the research and implementation done in [1], [2] by
+//! Marc Stevens.
+//!
+//!
+//! Original license from [3].
+//!
+//! > MIT License
```

Comment:
I don't think we need to include this licensing information in module-level docs. It can be included behind `//` comments. And `>`s are not needed as well.

Reply:
Since we've received permission from @cr-marcstevens to dual license as Apache 2.0+MIT (and per his comment it seems more of the code in the crate is a translation of that code?), I think we can provide some toplevel attribution (e.g. in README.md, which we can import into the rustdoc) mentioning that it contains translations of that code, the original copyright information, and that the resulting translation has been dual licensed Apache 2.0+MIT with permission.